src/conf_mode/protocols_rip.py

```diff
@@ -0,0 +1,313 @@
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from sys import exit
+
+from vyos import ConfigError
+from vyos.config import Config
+from vyos.util import call
+from vyos.template import render
+
+from vyos import airbag
+airbag.enable()
+
+config_file = r'/tmp/ripd.frr'
+
+def get_config():
+    conf = Config()
+    rip_conf = {
+        'rip_conf'          : False,
+        'default_distance'  : None,
+        'default_originate' : False,
+        'old_rip'  : {
+            'default_metric'  : None,
+            'distribute'      : {},
+            'neighbors'       : {},
+            'networks'        : {},
+            'net_distance'    : {},
+            'passive_iface'   : {},
+            'redist'          : {},
+            'route'           : {},
+            'ifaces'          : {},
+            'timer_garbage'   : 120,
+            'timer_timeout'   : 180,
+            'timer_update'    : 30
+        },
+        'rip'  : {
+            'default_metric'   : None,
+            'distribute'       : {},
+            'neighbors'        : {},
+            'networks'         : {},
+            'net_distance'     : {},
+            'passive_iface'    : {},
+            'redist'           : {},
+            'route'            : {},
+            'ifaces'           : {},
+            'timer_garbage'    : 120,
+            'timer_timeout'    : 180,
+            'timer_update'     : 30
+        }
+    }
+
+    if not (conf.exists('protocols rip') or conf.exists_effective('protocols rip')):
+        return None
+
+    if conf.exists('protocols rip'):
+        rip_conf['rip_conf'] = True
+
+    conf.set_level('protocols rip')
+
+    # Get default distance
+    if conf.exists_effective('default-distance'):
+        rip_conf['old_default_distance'] = conf.return_effective_value('default-distance')
+
+    if conf.exists('default-distance'):
+        rip_conf['default_distance'] = conf.return_value('default-distance')
+
+    # Get default information originate (originate default route)
+    if conf.exists_effective('default-information originate'):
+        rip_conf['old_default_originate'] = True
+
+    if conf.exists('default-information originate'):
+        rip_conf['default_originate'] = True
+
+    # Get default-metric
+    if conf.exists_effective('default-metric'):
+        rip_conf['old_rip']['default_metric'] = conf.return_effective_value('default-metric')
+
+    if conf.exists('default-metric'):
+        rip_conf['rip']['default_metric'] = conf.return_value('default-metric')
+
+    # Get distribute list interface old_rip
```

Comment:
I see you read in some parts of the config as `return_effective_value()` and some as `return_value()`. Why not cleanup the code and have a function reading in the config nodes? Thus `return_effective_value()` and `return_value()` can be selected via a lambda statement and you need to write the code only once -> eases maintenance.

Another idea would be to use the `config.get_config_dict()` function as on the long run everything should be ported there.